### ISSUE TITLE
DOMPDF 0.6.1+ on PHP 5.5.26/Mac, not working at all

### ISSUE THREAD

**Issue by makadev** [dompdf/dompdf#1055]
Hi,

To make it short, I've spend several hours on trying to get a usable PDF out of an HTML Document via DOMPDF 0.6.1/0.6.2-hotfix/0.7.0 beta2 - installed with composer - on PHP 5.5.26/Mac.

And the long description:
I have  to generate a one Page PDF Document from a simple Layout with a Background (given as Image or Header/Footer Image+Background Color) and some horizontal/vertical centered Text.
Since everything I need for that is Documented (http://pxd.me/dompdf/www/examples.php) as working, I did not expect it to.. Well, not working. My Time for testing and debugging 3rd Party tools is limited so I've only tested some setups (DOMPDF 0.6.1/0.6.2-hotfix/0.7.0 beta2) with mostly same results and did not go further into the generation process:

With the **CPDF** Backend there is no error or anything, it simply creates a PDF File that contains Text (unformatted)/Pictures behind one another. The only thing I could get to work was the resizing of images via width/height attributes on IMG Tag. Colors/Borders/Positions and pretty much everything else is ignored.
As an example:
https://github.com/dompdf/dompdf/blob/b09e8e9eb8c0007d61b4123783a62f34673c5237/www/test/css_position_absolute.html
and the generated Document [document-10.pdf](https://github.com/dompdf/dompdf/files/37719/document-10.pdf)

I've tried to use the **GD** Backend (on 0.7.0. beta2 directly by modifying the Adapter loading code since the Backend Option was kindly ignored) which ends in Font errors regardless of wether I set the full path to the lib/fonts directory or directly set it via Environmental Variabel GDFONTPATH. (Putting the fonts in the scripts root path is not an Option)

**PDFLib** is no option, those not tested.

Here is a list of installed php modules:
```
[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dba
dom
ereg
exif
fileinfo
filter
ftp
gd
gettext
hash
iconv
json
ldap
libxml
mbstring
mcrypt
mhash
mysql
mysqli
mysqlnd
odbc
openssl
pcntl
pcre
PDO
pdo_mysql
PDO_ODBC
pdo_pgsql
pdo_sqlite
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
SPL
sqlite3
standard
sysvmsg
sysvsem
sysvshm
tokenizer
wddx
xdebug
xml
xmlreader
xmlrpc
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Xdebug
Zend OPcache
```

Setup.php System Config (slightly modified for composer autoloader)
```
Required	Present
PHP Version	5.0	5.5.26
DOMDocument extension	Yes	20031129
PCRE	Yes	Yes
Zlib	Yes	2.0
MBString extension	Yes	Yes
GD	Yes	Yes
opcache	For better performances	Zend OPCache 7.0.6-devFE
GMagick or IMagick	Better with transparent PNG images	No. Recommended for better performances
```

Setup.php DOMPDF Config
```
DOMPDF_ENABLE_AUTOLOAD	false	Enable the DOMPDF autoloader	
DOMPDF_DIR	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf' Root directory of DOMPDF	Readable
DOMPDF_INC_DIR	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf/include'	Include directory of DOMPDF	Readable
DOMPDF_LIB_DIR	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf/lib'	Third-party libraries directory of DOMPDF	Readable
DOMPDF_ADMIN_USERNAME	******	The username required to access restricted sections	
DOMPDF_ADMIN_PASSWORD	******	The password required to access restricted sections	Password should be changed
DOMPDF_FONT_DIR	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf/lib/fonts/'	Directory containing fonts loaded into DOMPDF	Writable
DOMPDF_FONT_CACHE	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf/lib/fonts/'	Font metrics cache (used mainly by CPDF)	Writable
DOMPDF_TEMP_DIR	'/var/tmp/'	Temporary folder	Writable
DOMPDF_CHROOT	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf'	Restricted path	Readable
DOMPDF_UNICODE_ENABLED	true	Unicode support (with supporting fonts)	
DOMPDF_ENABLE_FONTSUBSETTING	false	Enable font subsetting, will make smaller documents when using Unicode fonts	
DOMPDF_PDF_BACKEND	'CPDF'	Backend library that renders the output (PDF, image)	
DOMPDF_DEFAULT_MEDIA_TYPE	'screen'	Default media type (print, screen, ...)	
DOMPDF_DEFAULT_PAPER_SIZE	'letter'	Default paper size (A4, letter, ...)	
DOMPDF_DEFAULT_FONT	'serif'	Default font, used if the specified font in the CSS stylesheet was not found	
DOMPDF_DPI	96	DPI scale of the document	
DOMPDF_ENABLE_PHP	false	Inline PHP support	
DOMPDF_ENABLE_JAVASCRIPT	true	Inline JavaScript support	
DOMPDF_ENABLE_REMOTE	false	Allow remote stylesheets and images	allow_url_fopen enabled
DOMPDF_LOG_OUTPUT_FILE	'/Users/makadev/Project/projectname/src/lib/composer/vendor/dompdf/dompdf/lib/fonts/log.htm'	The file in which dompdf will write warnings and messages	Not writable
DOMPDF_FONT_HEIGHT_RATIO	1.1000000000000001	The line height ratio to apply to get a render like web browsers	
DOMPDF_ENABLE_CSS_FLOAT	false	Enable CSS float support (experimental)	
DOMPDF_AUTOLOAD_PREPEND	false	Prepend the dompdf autoload function to the SPL autoload functions already registered instead of appending it	
DOMPDF_ENABLE_HTML5PARSER	false	Enable the HTML5 parser (experimental)	
DEBUGPNG	false	Debug PNG images	
DEBUGKEEPTEMP	false	Keep temporary image files	
DEBUGCSS	false	Debug CSS	
DEBUG_LAYOUT	false	Debug layout	
DEBUG_LAYOUT_LINES	true	Debug text lines layout	
DEBUG_LAYOUT_BLOCKS	true	Debug block elements layout	
DEBUG_LAYOUT_INLINE	true	Debug inline elements layout	
DEBUG_LAYOUT_PADDINGBOX	true	Debug padding boxes layout	
```

As far as I can see, the documented requirements are met (dom/mbstring/gd/php 5.3+). The only thing weird here is that DOMPDF_LOG_OUTPUT_FILE is shown as not writable thought this file does not have any different permissions set.
Anyway I'm pretty much done with DOMPDF and will switch to another document generator. I'm only adding this Ticket since I couldn't find anyone else with the same Problem, so maybe you can find the Problem/Solutions.

**Comment by bsweeney** [comment]
So at this point your *are not* looking for help in getting dompdf working? If you do decide to give dompdf another shot it would help to see a sample of the output plus the HTML you are attempting to render.

**Comment by makadev** [comment]
Hi,

I did invest another hour and am glad for it.. Short: The Problem was not in dompdf.

After setting up dompdf separately via composer, I could process the samples without problems.
So I reverted the Changes to the bigger Project and checked wether something manipulates the html. 

Simply said, there was a preprocessing step (small template engine) that processed the html and removed the opening braces for CSS declarations and maybe something else.

Sorry for the inconvenience.
